src/jdb_to_nwb/convert_photometry.py

```diff
@@ -1,22 +1,199 @@
 from pynwb import NWBFile
+import scipy.io # for loading in signals.mat files
+import pandas as pd # essential functions
+import numpy as np # essential functions
+from scipy.sparse import diags, eye, csc_matrix  # for creating sparse matrices
+from scipy.sparse.linalg import spsolve  # for solving sparse linear systems
+from sklearn.linear_model import Lasso # for Lasso regression
 
+def WhittakerSmooth(x,w,lambda_,differences=1):
+    '''
+    Penalized least squares algorithm for background fitting
+    
+    input
+        x: input data (i.e. chromatogram of spectrum)
+        w: binary masks (value of the mask is zero if a point belongs to peaks and one otherwise)
+        lambda_: parameter that can be adjusted by user. The larger lambda is,  the smoother the resulting background
+        differences: integer indicating the order of the difference of penalties
+    
+    output
+        the fitted background vector
+    '''
+    X=np.matrix(x)
+    m=X.size
+    i=np.arange(0,m)
+    E=eye(m,format='csc')
+    D=E[1:]-E[:-1] # numpy.diff() does not work with sparse matrix. This is a workaround.
+    W=diags(w,0,shape=(m,m))
+    A=csc_matrix(W+(lambda_*D.T*D))
+    B=csc_matrix(W*X.T)
+    background=spsolve(A,B)
+    return np.array(background)
+
+def airPLS(x, lambda_=100, porder=1, itermax=15):
+    '''
+    Adaptive iteratively reweighted penalized least squares for baseline fitting
+    
+    input
```

Comment:
Please add a link to a paper or other source that explains this algorithm in more detail (for others and your future self)